akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1933,6 +1933,22 @@ trait FlowOpsMat[+Out, +Mat] extends FlowOps[Out, Mat] {
     viaMat(GraphStages.terminationWatcher)(matF)
 
   /**
+    * Materializes to `FlowMonitor[Out]` that allows monitoring of the the current flow. All events are propagated
+    * by the monitor unchanged. Note that the monitor inserts a memory barrier every time it processes an
+    * event, and may therefor affect performance.
+    */
+  def monitor: ReprMat[Out, FlowMonitor[Out]] = monitorMat()(Keep.right)
```

Comment:
@drewhk, did we switch gears to allow combinators to have mat Keep.right by default? I remember we discussed this point at https://github.com/akka/akka/pull/19498#discussion_r50118558

>I see your point, but at some point we decided that the rules are:
Keep.right everywhere, except
runWith, foldWith, etc.
This makes the "matless" version not too useful though. would be interesting what others think

that's why watch watchTermination exists only with mat combinator as attribute.